# GEMMA from pangenome graph using BxD family of mouse

### 0. Extracted all linked-read folders'
```
/lizardfs/flaviav/mouse/assembly_D/GWAS/script/extract.sh
```
### 0. Build the pangenome graph whole-genome with pggb (using the near-T2T of C57BL/6J-DBA/2J-reference genome)

```
sbatch -p workers --job-name filterasvsref -c 48 --wrap "hostname; cd /scratch; pggb-a814fa58b370eddd5a099053f39d792bceab9258 -i /lizardfs/flaviav/mouse/assembly_D/8_pggb/D_C_mm10.fa.gz -n 3 -o /lizardfs/flaviav/mouse/assembly_D/GWAS/pggb_out_ref"
```

### 1. Mapping linked-read vs C57BL/6J + DBA/2J assembly
### 2. Inject the BAM (alignment) file from the above step into the pangenome graph
### 3. Generated a matrix of coverage

```
sbatch -p workers --job-name pregemma -c 1 --wrap "hostname; cd /scratch; /lizardfs/flaviav/mouse/assembly_D/GWAS/script/pregwas_all.sh"
```
### 4.  Merge all coverage matrices in a unique file (153 samples)

Merge the file follow the same order of the genomic standard approach
```
# Get the header from the first file
head -n 1 4512-JFI-0346_pack.tsv > matrcov153.tsv
file_list="samples.txt"
# Read the file names from the file_list.txt file into an array
mapfile -t ordered_files < "$file_list"

# Append the content from the files (excluding the header)
for file in "${ordered_files[@]}"; do
    if [ -f "${file}_pack.tsv.gz" ]; then
        zcat "${file}_pack.tsv.gz" | tail -n +2 >> matrcov153.tsv
    fi
done
```
Reheader the file with the same notation and order of the genomic standard file
```
input_file="matrcov153.tsv"
new_names_file="new_samples.txt"
output_file="matrcov153_ren.txt"
head -n 1 "$input_file" > "$output_file"
readarray -t new_names < "$new_names_file"
tail -n +2 "$input_file" | cut -f2- | paste -d $'\t' <(printf '%s\n' "${new_names[@]}") - >> "$output_file"
```
- Transpose matrix of coverage:
`/lizardfs/guarracino/bigspose/target/release/bigspose matrcov89.tsv matrcov89_r_trans.rust.tsv`

#### 4a. Adjust the genomic standard vcf file (we will use this as positive controls)
- Exclude the samples in which we are not interesting
- Rename the samples

#### 4b. Debug the matrix of coverage
- This is a RUST script that gives a summary of the coverage.
```
/lizardfs/flaviav/mouse/assembly_D/GWAS/script/debug_coverage/target/release/debug_coverage <norm_rescaled_1.csv> <norm_rescaled_2.csv> <output_file.tsv> <combined_rescaled_output.tsv> <coverage_summary_output.tsv> <sample_coverage_summary.tsv>
```
- This is a RUST script that generates a random coverage matrix to visualize it in R

- This is a python script that generates big coverage matrix (> 1000 and the samples' percentage)--> to do in RUST
`/bin/python3 /lizardfs/flaviav/mouse/assembly_D/GWAS/script/debug_big_cov.py matrcov89_r_trans.rust.tsv big_cov.txt

In R I upload all the above results to generate summary plots.
```
/lizardfs/flaviav/mouse/assembly_D/GWAS/script/debugmatr.R
```
### 5. Generating GEMMA's files. 

Steps to generating allele zygosity matrix.
- Generating coverage matrix with normalization filters:
I normalize the coverage values per sample, I divided each value in a row by the sum of values in that row. Then I had values between 0 and 1.
 `python3 /lizardfs/flaviav/mouse/assembly_D/GWAS/script/norm_1_2_matrixcov.py /lizardfs/flaviav/mouse/assembly_D/GWAS/out_gfainj/analyses/matrcov89_r_trans.rust.tsv /lizardfs/flaviav/mouse/assembly_D/GWAS/out_gfainj/analyses/in_gemma/norm_ 
`

#### 5b. GEMMA's inputs  

Using GN's file, put the phenotypes in the same order of the genotype file samples.

##### - Genotype file (called allele zygosity matrix)
We can try two different files.
Values' between 0 and 1.
I calculated the mean of all the values in the matrix. I divided each value in the matrix by the mean calculated. This will rescale the values so that their new mean is 1.0. After rescaling, I capped any values greater than 1.0 to 1.0.

```
/lizardfs/flaviav/mouse/assembly_D/GWAS/out_gfainj/analyses/in_gemma/norm_rescaled_1.csv
```

Values' between 0 and 2, this is the one that I am using.
I calculated the mean of all the values in the matrix. I divided each value in the matrix by the mean calculated. This will rescale the values so that their new mean is 2.0. After rescaling, I capped any values greater than 2.0 to 2.0.

```
/lizardfs/flaviav/mouse/assembly_D/GWAS/out_gfainj/analyses/in_gemma/norm_rescaled_2.csv
```

##### - Phenotype file 
Information about phenotype:

https://genenetwork.org/show_trait?trait_id=10358973&dataset=UTHSC_SPL_RMAEx_1210
```
/lizardfs/flaviav/mouse/assembly_D/GWAS/out_gfainj/analyses/in_gemma/pheno_89.txt
```

##### - Map file (called annotated file), we can use this for LOCO and for the GWAS plotting

Create a map file from the graph.This can help us to take tracks about the marker's position along the genome.
This is an experimental way, it will give to us all the possibile nodes and the localization on the chromosome. Can happens that different nodes are localized on different chromosomes. In addition we are using in that case the reference genome.

```
odgi-all-positions position -i graph.og --all-positions -R file_with_targets.txt
/lizardfs/flaviav/mouse/assembly_D/GWAS/matrixpos_allnodes.txt.gz
```
In addition seems GEMMA didn't like this format so we can do the following step:

zcat matrixpos_allnodes.txt.gz | sed 's/REF#chr//g' | awk 'NR==1 {next} {print $1, "node."$2, $3}' | awk -F" " '{print $2 "," $3 "," $1}' > annot_gemma.tsv.gz
#increment position of 1 awk -F',' '{$2=$2+1; print $1","$2","$3}' OFS=',' annt_gemma.tsv > annt_gemma_clean.txt


#### 5b. Run GEMMA
```
gemma -g norm_rescaled_2.csv -p pheno_89.txt -n 1 mouse_t2t_89_gk2.sXX.txt -notsnp -r2 1.0 -hwe 0 -miss 1.0 -lmm -o mouse_t2t_89_gk2.lmm.k1
gemma -g norm_rescaled_2.csv -p pheno_89.txt -n 1 -k mouse_t2t_89_gk2.sXX.txt -notsnp -r2 1.0 -hwe 0 -miss 1.0 -lmm -o mouse_t2t_89_gk2.lmm.k1
```
We are able to obtain the same peak that we expected on chr1 at the same location.

GeneNetwork
![Itvl_qYCNwPQB](https://github.com/Flavia95/PanGWAS/assets/52487106/006755bd-97cd-46d9-a595-b332ede24315)

Pangenome
![manhattan_final_pang](https://github.com/Flavia95/PanGWAS/assets/52487106/97bb8ac5-b0ea-4f2d-8585-a1d13b672df2)



###################  Additional steps

In addition we can create an annotation file, start-end bubbles with vg and cut only start and end. This can be use when we will not use the odgi position results anymore.

```
vg stats -R *final.vg > /home/flavia/work/DBA2J/GWAS/trait_154-155/start_end_only_wholegenome.tsv 
```

##### 6. Run GEMMA-wrapper (to do)

Gemma guide: https://www.xzlab.org/software/GEMMAmanual.pdf 


Gemma-wrapper allow running GEMMA with the LOCO option, it is useful when there is potential confounding due to population structure or relatedness. In the command line we specified also -lmm 9 that tells GEMMA to use the LOCO approach, where it performs association tests by leaving out one chromosome at a time from the analysis. This means that when testing for associations on a particular chromosome, the markers on that chromosome are not used to estimate the genetic relatedness matrix other analyses.

a. Kinship matrix, to obtain a Relatedness Matrix
b. Associations, run univariate LMM. Non-variant sites i.e. start/end of the bubbles are removed by GEMMA

```
gemma-wrapper --json --loco 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,X,Y,M -- -debug -g BXD.8_geno.txt -p pheno.txt -a BXD.8_snps.txt -gk > BXD_K_088C6Q.json
gemma-wrapper --json --loco --input BXD_K_088C6Q.json -- -debug -g BXD.8_geno.txt -p pheno.txt -a BXD.8_snps.txt -lmm 9 -maf 0.05 > BXD_GWA_3NMBLW.json
```
c. Example's output

```
chr     rs      ps      n_miss  allele1 allele0 af      beta    se      logl_H1 l_remle p_wald
-9      node.1  -9      0       A       A       0.325   -9.210579e-01   2.292064e+00    -4.395814e+01   8.026216e+00    6.925283e-01....
```
